--- zlib/1.2.13/CMakeLists.txt
+++ zlib/1.2.13/CMakeLists.txt
@@ -5,6 +5,8 @@ project(zlib C)
 
 set(VERSION "1.2.13")
 
+option(MINIZIP "Enable building MiniZip")
+
 set(INSTALL_BIN_DIR "${CMAKE_INSTALL_PREFIX}/bin" CACHE PATH "Installation directory for executables")
 set(INSTALL_LIB_DIR "${CMAKE_INSTALL_PREFIX}/lib" CACHE PATH "Installation directory for libraries")
 set(INSTALL_INC_DIR "${CMAKE_INSTALL_PREFIX}/include" CACHE PATH "Installation directory for headers")
@@ -120,6 +122,14 @@ set(ZLIB_SRCS
     zutil.c
 )
 
+if(MINIZIP)
+    set(ZLIB_SRCS ${ZLIB_SRCS}
+        contrib/minizip/zip.c
+        contrib/minizip/ioapi.c
+        contrib/minizip/unzip.c
+    )
+endif()
+
 if(NOT MINGW)
     set(ZLIB_DLL_SRCS
         win32/zlib1.rc # If present will override custom build rule below.
